### PR TITLE
enhanced identityDB

### PR DESCRIPTION
**Pull request by adecaro**
store in the identityDB also the identity configuration options and raw material that can be optionally used to load the key material from

### CODE REVIEW

File: token/services/db/sql/identity.go [hunk 1]
```diff
@@ -67,37 +67,49 @@ func NewIdentityDB(db *sql.DB, tablePrefix string, createSchema bool, signerInfo
...
+	)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed getting configuration for [%s:%s]", id, typ)
+	}
+	logger.Debugf("found configuration for [%s:%s]", id, typ)
+	return len(result) != 0, nil
+}
+
+func (db *IdentityDB) StoreIdentityData(id []byte, identityAudit []byte, tokenMetadata []byte, tokenMetadataAudit []byte) error {
+	query := fmt.Sprintf("INSERT INTO %s (identity_hash, identity, identityAuditInfo, tokenMetadata, tokenMetadataAuditInfo) VALUES ($1, $2, $3, $4, $5)", db.table.IdentityInfo)
```
Comment (alexandrosfilios) [inline token/services/db/sql/identity.go:100]: the naming of the columns is not consistent

File: token/services/db/sql/identity.go [hunk 2]
```diff
@@ -197,41 +242,47 @@ func (db *IdentityDB) signerInfoExists(h string) (bool, error) {
...
 		CREATE INDEX IF NOT EXISTS idx_ic_type_%s ON %s ( type );
+		CREATE INDEX IF NOT EXISTS idx_ic_id_type_%s ON %s ( id, type );
 
-		-- AuditInfo
+		-- IdentityInfo
 		CREATE TABLE IF NOT EXISTS %s (
             identity_hash TEXT NOT NULL PRIMARY KEY,
 			identity BYTEA NOT NULL,
-			info BYTEA NOT NULL
+			identityAuditInfo BYTEA NOT NULL,
```
Comment (alexandrosfilios) [inline token/services/db/sql/identity.go:283]: I'd use snake case for columns. they are case insensitive

File: token/services/identity/msp/idemix/lm.go [hunk 3]
```diff
@@ -360,6 +369,22 @@ func (lm *LocalMembership) loadFromStorage() error {
...
+			ID:   id,
+			Path: entry.URL,
+			Type: IdentityConfigurationType,
+		}
+		if len(entry.Config) != 0 {
+			if err := json.Unmarshal(entry.Config, identityConfig); err != nil {
+				logger.Errorf("failed to load configuration for entry [%s]", entry.ID)
+				continue
+			}
+			if identityConfig.ID != id || identityConfig.Path != entry.URL || identityConfig.Type != IdentityConfigurationType {
```
Comment (alexandrosfilios) [inline token/services/identity/msp/idemix/lm.go:383]: can this condition be false?

File: token/services/db/sql/identity.go (hunk 1 above)
Comment (adecaro) [reply token/services/db/sql/identity.go:100]: 👍

File: token/services/db/sql/identity.go (hunk 2 above)
Comment (adecaro) [reply token/services/db/sql/identity.go:283]: 👍

File: token/services/identity/msp/idemix/lm.go (hunk 3 above)
Comment (adecaro) [reply token/services/identity/msp/idemix/lm.go:383]: it might if the database is correpted.

**Review by alexandrosfilios — APPROVED** [review]

**Pull request merged**